web/js/naturalEvents/wv.naturalEvents.markers.js

```diff
@@ -0,0 +1,175 @@
+        pinEl.addEventListener(type, function(e){
+          willSelect = true;
+          passEventToTarget(e, olViewport);
+        });
+      });
+      ['pointermove', 'wheel', 'pointerdrag',
+        'pointerup', 'mousemove', 'touchmove'].forEach(function(type){
+        pinEl.addEventListener(type, function(e){
+          if (type === 'pointermove' || type === 'wheel') willSelect = false;
+          passEventToTarget(e, olViewport);
+        });
+      });
+      pinEl.addEventListener('click', function(e){
+        if (willSelect && !isSelected) {
+          ui.naturalEvents.selectEvent(event.id,date);
+        } else {
+          passEventToTarget(e, olViewport);
+        }
+      });
+      return marker;
+    });
+    map.renderSync(); // Marker position will be off until this is called
+    return markers;
+  };
+
+  self.remove = function(markers) {
+    markers = markers || [];
+    if (markers.length<1) return;
+    markers.forEach(function(marker){
+      if (marker.boundingBox) map.removeLayer(marker.boundingBox);
+      if (marker.pin) map.removeOverlay(marker.pin);
+    });
+  };
+
+  return self;
+};
+
+var passEventToTarget = function(event, target) {
+  try {
+    var eventCopy = new event.constructor(event.type, event);
+    target.dispatchEvent(eventCopy);
+  } catch (err) {} // Nothing to do, the try/catch prevents IE from crashing
+};
+
+var createPin = function(id, eventCategory, isSelected){
+  // Build SVG Element, using this instead of an img element allows styling with CSS
+  // Note xlink:href is required by Safari, href by other browsers
+  var wrapper = document.createElement('div');
+  var svgNS = 'http://www.w3.org/2000/svg';
+  var xlinkNS = 'http://www.w3.org/1999/xlink';
+  var svgEl = document.createElementNS(svgNS, 'svg');
+  var eventSymbol = document.createElementNS(svgNS, 'use');
+
+  eventSymbol.setAttribute('href', '#marker-' + eventCategory);
+  eventSymbol.setAttributeNS(xlinkNS, 'xlink:href', '#marker-' + eventCategory);
+  wrapper.setAttribute('class', 'marker marker-' + eventCategory);
+
+  if (isSelected) {
+    var pinSymbol = eventSymbol.cloneNode(true);
+    pinSymbol.setAttribute('href', '#marker-pin');
+    pinSymbol.setAttributeNS(xlinkNS, 'xlink:href', '#marker-pin');
+    wrapper.classList.add('marker-selected');
+    svgEl.setAttribute('width', 36);
+    svgEl.setAttribute('height', 42);
+    svgEl.appendChild(pinSymbol);
+  } else {
+    var dotSymbol = eventSymbol.cloneNode(true);
+    dotSymbol.setAttribute('href', '#marker-dot');
+    dotSymbol.setAttributeNS(xlinkNS, 'xlink:href', '#marker-dot');
+    svgEl.setAttribute('width', 25);
+    svgEl.setAttribute('height', 29);
+    svgEl.appendChild(dotSymbol);
+  }
+
+  svgEl.appendChild(eventSymbol);
+  wrapper.appendChild(svgEl);
+
+  // Create Overlay
+  return new ol.Overlay({
+    element: wrapper,
+    positioning: 'bottom-center',
+    id: id
+  });
+};
+
+var createBoundingBox = function(coordinates){
```

Comment:
CreateBoundingBox and createPin are now global. Will want to include these inside the main function.

Reply:
Good catch!  😬